src/Features/Core/Portable/EditAndContinue/CommittedSolution.cs

```diff
@@ -427,11 +452,16 @@ await TryGetMatchingSourceTextAsync(log, sourceText, sourceFilePath, currentDocu
         }
     }
 
-    public void CommitSolution(Solution solution)
+    public void CommitChanges(Solution solution, ImmutableArray<ProjectId> projectsToStale, ImmutableArray<ProjectId> projectsToUnstale)
     {
+        Contract.ThrowIfFalse(projectsToStale is [] || projectsToUnstale is []);
+
         lock (_guard)
         {
             _solution = solution;
+            _staleProjects.AddRange(projectsToStale);
+            _staleProjects.RemoveRange(projectsToUnstale);
+            _documentState.Remove(static (documentId, _, projectsToUnstale) => projectsToUnstale.Contains(documentId.ProjectId), projectsToUnstale);
```

Comment:
on the fence if i want this called Remove/RemoveAll.  The problem with 'Remove' is that it conveys to me that it's removing the first thing that matches the predicate.  RemoveALl at least tells me that it's acutally all the matches.

Reply:
RemoveRange?

Reply:
i *think* i could live with that.  though 'Range' tends to me "i'll give you mutliple things to operate on".  RemoveAll just reads *right* to me.

Reply:
FYI: https://github.com/dotnet/dotnet-api-docs/issues/8177#issuecomment-1167251052. 

I'll change the extension to only apply to Dictionary.

Reply:
Re naming, RemoveAll it is: https://learn.microsoft.com/en-us/dotnet/api/system.collections.generic.list-1.removeall?view=net-9.0